### ISSUE TITLE
App Center (Ubuntu 24.04) not working behind corporate proxy

### ISSUE THREAD

**Issue by Langohr23** [ubuntu/app-center#1668]
Hello together,

I have setup Ubuntu 24.04. at my company PC. The network needs a corporate proxy (non intercepting, just a forwarding proxy). Everything works fine (APT in the terminal, other applications in the terminal, Browsers etc.) except the App Center.

I have configured our proxy in the following locations:
1. In the operating systems Network -> Proxy tap. First via a WPAD file, but I also tried it with a manuel configuration.
2. For APT in : /etc/apt/apt.conf.d/ (http, ftp and https)
3. For the terminal in "etc/enviornment" (HTTP and HTTPS)

I always get the following error in the App center:
![image](https://github.com/ubuntu/app-center/assets/33636936/838e3779-4324-4815-b571-3957dcfc42dd)

In the Logs I get the following error: 

SnapdException: SnapdException(kind: null, message: cannot list updates: Post "https://api.snapcraft.io/v2/snaps/refresh": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers))

Thank you for you help!

Thomas

**Comment by d-loose** [comment]
Thanks for your report! Can you successfully interact with snapd on the command line? Are there any errors if you run `snap install hello` or `snap refresh`? If that's the case I'd point you in the direction of the [snapcraft forums](https://forum.snapcraft.io/) for support, as we don't have any influence on that in the frontend.

In any case I think we need a more graceful error handling here.

**Comment by SergeyNovak777** [comment]
Hello, you can trying write proxy for snap
```
sudo snap set system proxy.http="your_proxy:port"
sudo snap set system proxy.https="your_proxy:port"
```
in my work pc:
```
sudo snap set system proxy.http="http://proxy.unn.ru:8080"
sudo snap set system proxy.https="http://proxy.unn.ru:8080"
```

**Comment by d-loose** [comment]
Error handling of this issue is addressed in #1723
